arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Iterable.kt

```diff
@@ -347,10 +398,20 @@ public inline fun <A, B> Iterable<A>.traverseOption(f: (A) -> Option<B>): Option
   return destination.some()
 }
 
+@Deprecated("use sequence instead", ReplaceWith("sequence()", "arrow.core.sequence"))
 public fun <A> Iterable<Option<A>>.sequenceOption(): Option<List<A>> =
-  this.traverseOption { it }
+  sequence()
+
+public fun <A> Iterable<Option<A>>.sequence(): Option<List<A>> =
+  traverse(::identity)
 
-public inline fun <A, B> Iterable<A>.traverseNullable(f: (A) -> B?): List<B>? {
+@Deprecated("use traverse instead", ReplaceWith("traverse(f)", "arrow.core.traverse"))
+public inline fun <A, B> Iterable<A>.traverseNullable(f: (A) -> B?): List<B>? =
+  traverse(f)
+
+@OptIn(ExperimentalTypeInference::class)
+@OverloadResolutionByLambdaReturnType
+public inline fun <A, B> Iterable<A>.traverse(f: (A) -> B?): List<B>? {
```

Comment:
Curious that this doesn't conflict with the other `traverse` functions 🤔
This seems like one of those signatures that breaks all other functions 😅

What happens in the case of:

```kotlin
fun either(i: Int): Either<String, Int>? = null

val x: ??? = listOf(1, 2, 3).traverse { either(i) }
```

I guess `???` here is `List<Either<String, Int>>?`, and to turn it into `Either<String, List<Int>>?` we'd need to chain it with `?.sequence()`. That'd be pretty cool if that works like that out of the box.

Reply:
Yes this one would be List<Either<String, Int>>? I can set up a small example test